src/js/heuristicblocking.js

```diff
@@ -499,44 +499,9 @@ function hasCookieTracking(details, origin) {
   return false;
 }
 
-function startListeners() {
-  /**
-   * Adds heuristicBlockingAccounting as listened to onBeforeSendHeaders request
-   */
-  chrome.webRequest.onBeforeSendHeaders.addListener(function(details) {
-    if (badger) {
-      return badger.heuristicBlocking.heuristicBlockingAccounting(details);
-    } else {
-      return {};
-    }
-  }, {urls: ["<all_urls>"]}, ["requestHeaders"]);
-
-  /**
-   * Adds onResponseStarted listener. Monitor for cookies
-   */
-  chrome.webRequest.onResponseStarted.addListener(function(details) {
-    var hasSetCookie = false;
-    for(var i = 0; i < details.responseHeaders.length; i++) {
-      if(details.responseHeaders[i].name.toLowerCase() == "set-cookie") {
-        hasSetCookie = true;
-        break;
-      }
-    }
-    if(hasSetCookie) {
-      if (badger) {
-        return badger.heuristicBlocking.heuristicBlockingAccounting(details);
```

Comment:
If I understand correctly, this situation would only be possible in the following circumstance:
- A user visits `bar.com`, and `foo.com` is loaded as a third party.
- `foo.com` sets a third-party cookie.
- While navigating `bar.com`, no more requests to `foo.com` are made (as any requests would by default include the cookie that was set).
- The user then navigates to `foo.com`, sending the cookie that was set when `foo.com` was loaded as a third party.

It seems that in the worst case scenario a strike against `foo.com` would be avoided and `foo.com` would gain access to at most one piece of information: the site/page that `foo.com`'s cookie was set on. This is possible, but seems quite improbable as it would require `foo.com` to only be loaded once in that domain. Does this make sense? Am I overlooking something?